nltk/test/lm.doctest

```diff
@@ -129,3 +129,19 @@ For speed take only the first 100 sentences of reuters. Shouldn't affect the tes
     >>> lm.fit(train_data, vocab_data)
     >>> lm.score("said", ("",)) < 1
     True
+
+
+
+
+Issue 2562
+---------
+https://github.com/nltk/nltk/issues/2562
```

Comment:
To think of it, this isn't really a bug with the language model module, right? All the lm code works with both sequences and iterables iirc and is documented as such. Would it be clearer if we add this regression tests to the `utils.doctest` file?

Reply:
I wasn't sure where to add the tests. I am happy to move them where ever you think is appropriate. Are you referring to the `util.doctest` file in the test folder? It could be beneficial to add tests for iterables with other related functions as well. What are your thoughts on this?

Most of the `*grams` functions (`ngrams`, `bigrams`, etc.) work with iterables, but the current implementation of everygrams assumes that the iterable has a `len` function defined (if `max_len` is not given) and potentially depletes the iteratable on the first call to `ngrams`.

```python
>>> list(everygrams(sent))
[('a',), ('b',), ('c',), ('a', 'b'), ('b', 'c'), ('a', 'b', 'c')]
>>> list(everygrams(iter(sent)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../python3.8/site-packages/nltk/util.py", line 603, in everygrams
    max_len = len(sequence)
TypeError: object of type 'list_iterator' has no len()
>>> list(everygrams(sent, max_len=2))
[('a',), ('b',), ('c',), ('a', 'b'), ('b', 'c')]
>>> list(everygrams(iter(sent), max_len=2))
[('a',), ('b',), ('c',)]
```

Reply:
I'm honestly leaning towards adding a `tests/unit/test_utils.py` file. I have a mind to move those `lm` regression tests out of the `doctest` file, because the formatting forces us to overspecify things like the order of sequences even though we don't really care about it. You've run into this already with the whole need for `sorted`... In a unit test that would be done with `assertCountEqual`.

Reply:
IMHO, both doctest and unittest should exists, though it seem redundant. Partly because `help()` will show the docs when users want to learn more about it.

Reply:
Agreed! I'm just wondering if someone who calls `help()` is interested in reading the regression tests :)

Reply:
Should I add @iliakur's suggested `tests/unit/test_utils.py` file and move the tests there?

Reply:
Yes please! Make sure to name it `test_util.py` though, to stay consistent with the name of the module you're testing.